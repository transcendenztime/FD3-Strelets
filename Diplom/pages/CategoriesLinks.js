import React from 'react';
import { NavLink } from 'react-router-dom';

import '../styles/pages/CategoryLink.css';

class CategoriesLinks extends React.Component {
          
  render() {

    return (
      <ul className = "category-menu main-menu">
        <li><NavLink to="/catalogue/semena-ovoshchey" className="CategoryLink PageLink">Семена овощей</NavLink></li>
        <li><NavLink to="/catalogue/semena-cvetov" exact className="CategoryLink PageLink">Семена цветов</NavLink></li>
        <li><NavLink to="/catalogue/sazhency-roz" className="CategoryLink PageLink">Саженцы роз</NavLink></li>
        <li><NavLink to="/catalogue/lukovicy-i-klubnelukovicy-cvetov" className="CategoryLink PageLink">Луковицы и клубнелуковицы цветов</NavLink></li>
        <li><NavLink to="/catalogue/plodovye-derevya-i-kustarniki" className="CategoryLink PageLink">Плодовые деревья и кустарники</NavLink></li>
        <li><NavLink to="/catalogue/hvoynye-rasteniya" className="CategoryLink PageLink">Хвойные растения</NavLink></li>
		    <li><NavLink to="/catalogue/dekorativnye-kustarniki" className="CategoryLink PageLink">Декоративные кустарники</NavLink></li>
		    <li><NavLink to="/catalogue/rododendrony" className="CategoryLink PageLink">Рододендроны</NavLink></li>
      </ul>
    );

  }

}
    
export default CategoriesLinks;