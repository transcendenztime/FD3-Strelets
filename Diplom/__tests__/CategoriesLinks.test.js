import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import CategoriesLinks from '../pages/CategoriesLinks';

test('внешний вид CategoriesLinks', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
      <Router>
          <CategoriesLinks />
      </Router>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
          
  });