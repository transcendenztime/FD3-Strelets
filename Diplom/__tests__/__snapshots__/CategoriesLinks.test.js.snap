// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид CategoriesLinks 1`] = `
<ul
  className="category-menu main-menu"
>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/semena-ovoshchey"
      onClick={[Function]}
      style={undefined}
    >
      Семена овощей
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/semena-cvetov"
      onClick={[Function]}
      style={undefined}
    >
      Семена цветов
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/sazhency-roz"
      onClick={[Function]}
      style={undefined}
    >
      Саженцы роз
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/lukovicy-i-klubnelukovicy-cvetov"
      onClick={[Function]}
      style={undefined}
    >
      Луковицы и клубнелуковицы цветов
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/plodovye-derevya-i-kustarniki"
      onClick={[Function]}
      style={undefined}
    >
      Плодовые деревья и кустарники
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/hvoynye-rasteniya"
      onClick={[Function]}
      style={undefined}
    >
      Хвойные растения
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/dekorativnye-kustarniki"
      onClick={[Function]}
      style={undefined}
    >
      Декоративные кустарники
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="CategoryLink PageLink"
      href="/catalogue/rododendrony"
      onClick={[Function]}
      style={undefined}
    >
      Рододендроны
    </a>
  </li>
</ul>
`;
